### PR TITLE
Update Dockerfile to use ubuntu 16.04

### PR DESCRIPTION
**Pull request by jingyimei**
This commit adds a new dockerfile to bake postgres 10.5 on ubuntu
16.04. Also updates docker_start.sh and README to pull the new docker image instead
of the old one (Postgres9.6 on Ubuntu 8.9).

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/703/

**Comment by orhankislal** [comment]
Please do not merge this PR until we change the version to 1.16-dev.

File: tool/docker/base/Dockerfile_ubuntu16_postgres10 [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+      cd build && \
+      cmake -G "Unix Makefiles" .. && \
+      make && \
+      make install
+
+### Optional: install LaTex
+### uncomment the following 'RUN apt-get' line to bake LaTex into the image
+### Note: if you run the following line, please tag the image as
+### madlib/postgres_10:LaTex, and don't tag it as latest
+# RUN apt-get install -y texlive-full
```
[the remark this answers is not in the document]
Comment (jingyimei) [reply tool/docker/base/Dockerfile_ubuntu16_postgres10]: This is for making design doc (pdf). LaTex is pretty big and thus we make different images so people can pull it when needed.

File: tool/docker/base/Dockerfile_ubuntu16_postgres10 [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+      tar -zxvf cmake-3.6.1.tar.gz && \
+      cd cmake-3.6.1 && \
+      sed -i 's/-DCMAKE_BOOTSTRAP=1/-DCMAKE_BOOTSTRAP=1 -DCMAKE_USE_OPENSSL=ON/g' bootstrap && \
+      ./configure &&  \
+      make -j2 && \
+      make install && \
+      cd ..
+
+### Install doxygen-1.8.13:
+RUN wget http://ftp.stack.nl/pub/users/dimitri/doxygen-1.8.13.src.tar.gz && \
```
[the remark this answers is not in the document]
Comment (jingyimei) [reply tool/docker/base/Dockerfile_ubuntu16_postgres10]: `make doc` is optional when people want to install madlib, and that's why I removed the line `make doc` before. But I think it is OK to bake doxygen in the image and people can run `make doc` when needed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/706/

**Pull request closed without merging**